src/misc/namespaced_id.rs

```diff
@@ -13,7 +13,13 @@ pub struct NamespacedId {
 
 impl NamespacedId {
     /// Create a new NamespacedId
-    pub fn new(is_tag: bool, namespace: &str, id: &str) -> Result<Self, InvalidNamespacedIdError> {
+    pub fn new<S: Into<String>>(
```

Comment:
Any particular reason to change this to take Intos?

Reply:
It just makes the interface a bit nicer and prevents double allocation when creating a new `NamespacedId` with a new `String`.